daemon/cmd/kube_proxy_replacement.go

```diff
@@ -42,7 +43,7 @@ import (
 //
 // if this function cannot determine the strictness an error is returned and the boolean
 // is false. If an error is returned the boolean is of no meaning.
-func initKubeProxyReplacementOptions() error {
+func initKubeProxyReplacementOptions(tunnelConfig tunnel.Config) error {
```

Comment:
I see reasons to pass tunnelConfig as a parameter, and I understand there already have been a `mtuConfig` under `Daemon` struct so it makes sense to put `tunnelConfig` in the same place. Still, this sets a precedent to pass config everywhere instead of using the global `option.Config`. It's not this PR's fault, and I don't have a easy solution either. Maybe we can consider moving these configs into option pkg in future, while it goes beyond this PR's scope.

Reply:
My understanding is that the long term goal is to get rid as much as possible of the global `option.Config` object, and move the various settings close to the corresponding cell. This is what I've been seeing as part of the refactoring of various modules to the hive/cell paradigm, e.g., \[1\]. IMO, passing config objects as parameter has the advantage of making their usage more explicit, rather than just accessing a global variable/function deep in the code path (although at the cost of having to propagate it), and simplify at the same time unit testing, as we depend less on global state.

Now, I agree that putting `tunnelConfig` under `Daemon` is not great. Ideally, consumers (like the KPR module in this case) should also be cells, and could just depend on the `tunnelConfig` itself. The current approach is mostly a tradeoff to limit the scope of the single refactoring. Long term, these variables should be removed from `Daemon`.

\[1\]: https://github.com/cilium/cilium/pull/28619#discussion_r1381395461